activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -243,7 +243,9 @@ def format_options(options)
       end
 
       def remove_prefix_and_suffix(table)
-        table.gsub(/^(#{@options[:table_name_prefix]})(.+)(#{@options[:table_name_suffix]})$/,  "\\2")
+        prefix = Regexp.escape(@options[:table_name_prefix].to_s)
+        suffix = Regexp.escape(@options[:table_name_suffix].to_s)
+        table.sub(/\A(?:#{prefix})(.+)(?:#{suffix})\z/, "\\1")
```

Comment:
Thank you. Does this line make sense to you? At least unit test successfully finished with this change.

```
 table.sub(/\A#{prefix}(.+)#{suffix}\z/, "\\1")
```